src/main/scala/mesosphere/marathon/api/v2/json/Formats.scala

```diff
@@ -239,13 +239,47 @@ trait ContainerFormats {
 trait IpAddressFormats {
   import Formats._
 
-  implicit lazy val NetworkProtocolFormat: Format[mesos.NetworkInfo.Protocol] =
-    enumFormat(mesos.NetworkInfo.Protocol.valueOf, str => s"$str is not a valid protocol")
+  private[this] lazy val ValidPortProtocol: Reads[String] = {
+    implicitly[Reads[String]]
+      .filter(ValidationError("Invalid protocol. Only 'udp' or 'tcp' are allowed."))(
```

Comment:
Mesos proto is not strict here. But I don't see any reason why not to be strict here.